internal/ethapi/api.go

```diff
@@ -764,6 +770,43 @@ func (s *BlockChainAPI) GetProof(ctx context.Context, address common.Address, st
 	}, statedb.Error()
 }
 
+// isAllowedToBeQueried returns a nil error only if either:
+//   - the node is configured to accept historical proofs queries; or
+//   - the block number given is within the window of recent accepted
+//     blocks and not yet part of the historical blocks.
+//
+// Otherwise, a non-nil error is returned.
+func (s *BlockChainAPI) isAllowedToBeQueried(blockNumOrHash rpc.BlockNumberOrHash) (err error) {
```

Comment:
let's make all additions. in files that are not shared with upstream code such as `internal/ethapi/api_extra.go`. This will help us handle upstream merges easier.
The same idea applies to test code as well.

Reply:
My apologies I did not check, I will in the future 👍 
Moved to api_extra.go and api_extra_test.go.